### PR TITLE
Install dependencies via meta package

### PR DESCRIPTION
**Pull request by raydouglass**
Install dependencies via `rapids-build-env` (which is pre-installed in the containers).

This brings the repo in line with https://docs.rapids.ai/maintainers/depmgmt/

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: ci/gpu/build.sh [hunk 1]
```diff
@@ -44,7 +44,12 @@ nvidia-smi
 
 logger "Activate conda env..."
 source activate gdf
-conda install "cudf=${MINOR_VERSION}.*" "cudatoolkit=$CUDA_REL" "gdal>=3.0.2"
+conda install "cudf=${MINOR_VERSION}.*" "cudatoolkit=$CUDA_REL" \
+    "rapids-build-env=$MINOR_VERSION.*"
```
Comment (thomcom) [inline ci/gpu/build.sh:48]: What happens to the gdal dependency in this change?

Comment (raydouglass) [reply ci/gpu/build.sh:48]: It's in `rapids-build-env`

Comment (thomcom) [reply ci/gpu/build.sh:48]: Great, where is it? How can I take a look?

Comment (ajschmidt8) [reply ci/gpu/build.sh:48]: hey @thomcom. we keep the conda environments in this repo: https://github.com/rapidsai/integration/

`rapids-build-env` specifically is located here https://github.com/rapidsai/integration/blob/branch-0.15/conda/recipes/rapids-build-env/meta.yaml

Comment (raydouglass) [reply ci/gpu/build.sh:48]: https://github.com/rapidsai/integration/blob/branch-0.15/conda/recipes/rapids-build-env/meta.yaml#L70

**Pull request merged**